arelle/Cntlr.py

```diff
@@ -265,8 +312,8 @@ def setUiLanguage(self, lang, fallbackToDefault=False):
             self.uiLang = langCodes[0].lower()
             self.uiLangDir = 'rtl' if self.uiLang[0:2] in {"ar","he"} else 'ltr'
             if not isPy3: # 2.7 gettext provides string instead of unicode from .mo files
-                installedGettext = __builtins__['_']
-                def convertGettextResultToUnicode(msg):
+                installedGettext: TypeGetText = __builtins__['_']
+                def convertGettextResultToUnicode(msg: str) -> Any:
                     translatedMsg = installedGettext(msg)
                     if isinstance(translatedMsg, _STR_UNICODE):
```

Comment:
Add `# type: ignore[name-defined]` here and in all places where you get the error `name-defined`

Reply:
Got it.
On another note, I think these lines should be removed as a whole, there are for python 2.7, not supported anymore

Reply:
Maybe something to consider in this issue? https://github.com/Arelle/Arelle/issues/309

Reply:
removed py 2 code fd85d6ba4abe2ad1106765eb64726f490de58ac0